interface/forms/soap/templates/general_new.html

```diff
@@ -1,52 +1,58 @@
+                                    </div>
+                                </div>
+                    </fieldset>
+                    <fieldset>
+                            <legend>{xl t='Objective'|escape:'html'}</legend>
+                                <div class="form-group">
+                                    <div class="col-sm-10 col-sm-offset-1">
+                                        <textarea name="objective" class="form-control" cols="60" rows="6">{$data->get_objective()|escape:'html'}</textarea>
+                                    </div>
+                                </div>
+                    </fieldset>
+                    <fieldset>
+                            <legend>{xl t='Assessment'|escape:'html'}</legend>
+                                <div class="form-group">
+                                    <div class="col-sm-10 col-sm-offset-1">
+                                        <textarea name="assessment" class="form-control" cols="60" rows="6">{$data->get_assessment()|escape:'html'}</textarea>
+                                    </div>
+                                </div>
+                    </fieldset>
+                    <fieldset>
+                            <legend>{xl t='Plan'|escape:'html'}</legend>
+                                <div class="form-group">
+                                    <div class="col-sm-10 col-sm-offset-1">
+                                        <textarea name="plan" class="form-control" cols="60" rows="6">{$data->get_plan()|escape:'html'}</textarea>
+                                    </div>
+                                </div>
+                    </fieldset>
+                    <?php //can change position of buttons by creating a class 'position-override' and adding rule text-align:center or right as the case may be in individual stylesheets ?>
+                    <div class="form-group clearfix">
+                        <div class="col-sm-10 col-sm-offset-1 position-override">
+                            <div class="btn-group oe-opt-btn-group-pinch" role="group">
                                 <button type="submit" class="btn btn-default btn-save" name="Submit">{xl t='Save'|escape:'html'}</button>
-                                <a href="{$DONT_SAVE_LINK}" class="btn btn-link btn-cancel" onclick="top.restoreSession()">{xl t='Cancel'|escape:'html'}</a>
+                                <button type="button" class="btn btn-link btn-cancel oe-opt-btn-separate-left"onclick="top.restoreSession(); location.href='{$DONT_SAVE_LINK}';">{xl t='Cancel'|escape:'html'}</button>
                             </div>
                             <input type="hidden" name="id" value="{$data->get_id()|escape:'html'}" />
                             <input type="hidden" name="activity" value="{$data->get_activity()|escape:'html'}">
```

Comment:
Just some minor issues on this form. Otherwise, this is testing well. After address these issues can them focus on the them issues previously discussed and then get the chunk of code into the codebase (even with just SOAP, this initial commit will be a big deal considering the ugprading of new themes and removal of old themes :)